portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,38 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> messages;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<>();
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
+    
+    response.setContentType("application/json");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    messages.add(request.getParameter("comment-input"));
```

Comment:
Use class constant variables for the hard coded Strings here.

Reply:
Why is using a constant variable preferable?

Reply:
It's generally best practice to keep your constants as private static final variables at the top of the file so that you can find and change them easily.

Reply:
Ahh, that makes sense. How come we don't do the same for the other Strings such as the "/#comments" on line 49?

Reply:
You should do it for that too, and your constant variables should be "private static final" and use all uppercase with underscores between words.  So something like:
private static final String REQUEST_PARAMETER_INPUT = "comment-input".

Can you see this? https://engdoc.corp.google.com/eng/doc/devguide/java/style/index.md?cl=head#s5.2.4-constant-names

Reply:
I unfortunately cannot see it